pkg/clients/database/sql.go

```diff
@@ -0,0 +1,612 @@
+	nameClient.Authorizer = c.Authorizer
+	nameClient.AddToUserAgent(azure.UserAgent)
+
+	return &PostgreSQLServerClient{
+		ServersClient:               postgreSQLServerClient,
+		CheckNameAvailabilityClient: nameClient,
+	}, nil
+}
+
+// ServerNameTaken returns true if the supplied server's name has been taken.
+func (c *PostgreSQLServerClient) ServerNameTaken(ctx context.Context, s *azuredbv1alpha3.PostgreSQLServer) (bool, error) {
+	r, err := c.Execute(ctx, postgresql.NameAvailabilityRequest{Name: azure.ToStringPtr(meta.GetExternalName(s))})
+	if err != nil {
+		return false, err
+	}
+	return !azure.ToBool(r.NameAvailable), nil
+}
+
+// GetServer retrieves the requested PostgreSQL Server
+func (c *PostgreSQLServerClient) GetServer(ctx context.Context, s *azuredbv1alpha3.PostgreSQLServer) (postgresql.Server, error) {
+	return c.ServersClient.Get(ctx, s.Spec.ForProvider.ResourceGroupName, meta.GetExternalName(s))
+}
+
+// CreateServer creates a PostgreSQL Server
+func (c *PostgreSQLServerClient) CreateServer(ctx context.Context, cr *azuredbv1alpha3.PostgreSQLServer, adminPassword string) error {
+	s := cr.Spec.ForProvider
+	properties := &postgresql.ServerPropertiesForDefaultCreate{
+		AdministratorLogin:         azure.ToStringPtr(s.AdministratorLogin),
+		AdministratorLoginPassword: &adminPassword,
+		Version:                    postgresql.ServerVersion(s.Version),
+		SslEnforcement:             postgresql.SslEnforcementEnum(s.SSLEnforcement),
+		CreateMode:                 postgresql.CreateModeDefault,
+		StorageProfile: &postgresql.StorageProfile{
+			BackupRetentionDays: azure.ToInt32PtrFromIntPtr(s.StorageProfile.BackupRetentionDays),
+			GeoRedundantBackup:  postgresql.GeoRedundantBackup(azure.ToString(s.StorageProfile.GeoRedundantBackup)),
+			StorageMB:           azure.ToInt32Ptr(s.StorageProfile.StorageMB),
+			StorageAutogrow:     postgresql.StorageAutogrow(azure.ToString(s.StorageProfile.StorageAutogrow)),
+		},
+	}
+	sku, err := ToPostgreSQLSKU(s.SKU)
+	if err != nil {
+		return err
+	}
+	createParams := postgresql.ServerForCreate{
+		Sku:        sku,
+		Properties: properties,
+		Location:   &s.Location,
+		Tags:       azure.ToStringPtrMap(s.Tags),
+	}
+	op, err := c.Create(ctx, s.ResourceGroupName, meta.GetExternalName(cr), createParams)
+	if err != nil {
+		return err
+	}
+	_, err = op.DoneWithContext(ctx, c.ServersClient.Client)
+	return err
+}
+
+// UpdateServer updates a PostgreSQL Server.
+func (c *PostgreSQLServerClient) UpdateServer(ctx context.Context, cr *azuredbv1alpha3.PostgreSQLServer) error {
+	// TODO(muvaf): password update via Update call is supported by Azure but
+	// we don't support that.
+	s := cr.Spec.ForProvider
+	properties := &postgresql.ServerUpdateParametersProperties{
+		Version:        postgresql.ServerVersion(s.Version),
+		SslEnforcement: postgresql.SslEnforcementEnum(s.SSLEnforcement),
+		//ReplicationRole: s.Spec.ForProvider.ReplicationRole,
```

Comment:
Should this be commented out?

Reply:
See https://github.com/crossplaneio/stack-azure/pull/64#discussion_r345739108

Reply:
```suggestion
		// ReplicationRole is not used in DefaultCreateMode.
		//ReplicationRole: s.Spec.ForProvider.ReplicationRole,
```